bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -733,6 +735,10 @@ public synchronized void close() throws IOException {
       throw new IOException("Could not close the admin client", ex);
     }
 
+    if (options.useCachedChannel() && options.useGCJClient()) {
+      cachedClientContexts.remove(options.getDataHost());
```

Comment:
As discussed offline, this isn't enough. The background resources in the ClientContext need to be shutdown